junit-platform-launcher/src/main/java/org/junit/platform/launcher/TestIdentifier.java

```diff
@@ -216,4 +272,75 @@ public String toString() {
 		// @formatter:on
 	}
 
+	private void writeObject(ObjectOutputStream s) throws IOException {
+		SerializedForm serializedForm = new SerializedForm(this);
+		serializedForm.serialize(s);
+	}
+
+	private void readObject(ObjectInputStream s) throws ClassNotFoundException, IOException {
+		SerializedForm serializedForm = SerializedForm.deserialize(s);
+		uniqueId = UniqueId.parse(serializedForm.uniqueId);
+		displayName = serializedForm.displayName;
+		source = serializedForm.source;
+		tags = serializedForm.tags;
+		type = serializedForm.type;
+		parentId = UniqueId.parse(serializedForm.parentId);
+		legacyReportingName = serializedForm.legacyReportingName;
+	}
+
+	/**
+	 * Represents the serialized output of {@code TestIdentifier}. The fields on this
+	 * class match the files that {@code TestIdentifier} had prior to 5.8.
+	 */
+	private static class SerializedForm implements Serializable {
+
+		private static final long serialVersionUID = 1L;
+
+		private final String uniqueId;
+		private final String parentId;
+		private final String displayName;
+		private final String legacyReportingName;
+		private final TestSource source;
+		private final Set<TestTag> tags;
+		private final Type type;
+
+		public SerializedForm(TestIdentifier testIdentifier) {
```

Comment:
No need to make the constructor and methods `public` in this local class.

Reply:
Done in ffbafa654d153fbef048e6e56a6fd3786a44db08.